backend/src/infra/repository/GroupRepositoryInMemory.ts

```diff
@@ -0,0 +1,73 @@
+import { Group } from "src/core/entity/Group";
+import { type GroupMember } from "src/core/entity/GroupMember";
+import { Transaction } from "src/core/entity/Transaction";
+import { type GroupRepository } from "src/core/repository/GroupRepository";
+
+export class GroupRepositoryInMemory implements GroupRepository {
+  private readonly groups: Group[];
+  private nextId: number;
+  private nextTransacationId: number;
+
+  constructor(initialGroups?: Group[]) {
+    this.groups = initialGroups ?? [];
+    this.nextId =
+      initialGroups != null
+        ? initialGroups.reduce(
+            (currentId, group) => Math.max(currentId, group.id),
+            -1
+          ) + 1
+        : 0;
+    let nbTransactions = 0;
+    let maxTransactionId = 0;
+    if (initialGroups != null) {
+      for (const group of initialGroups) {
+        nbTransactions += group?.transactions.length;
+        for (const transaction of group.transactions) {
+          maxTransactionId = Math.max(maxTransactionId, transaction.id);
+        }
+      }
+    }
+    this.nextTransacationId = Math.max(
+      nbTransactions + 1,
+      maxTransactionId + 1
+    );
```

Comment:
```suggestion
const transactionIds = this.groups.flatMap(group => group.transaction).map(transaction => transaction.id);
this.nextTransactionId = transactionIds.length === 0 ? 0 : Math.max(...transactionIds) + 1;
```

J'aurais du faire ça aussi au lieu du `reduce`, ça aurait été plus clair x)